cmd/kube-proxy/app/server.go

```diff
@@ -619,6 +622,9 @@ func (s *ProxyServer) Run() error {
 			err := http.ListenAndServe(s.MetricsBindAddress, proxyMux)
 			if err != nil {
 				utilruntime.HandleError(fmt.Errorf("starting metrics server failed: %v", err))
+				if s.MetricsBindHardFail {
+					errCh <- err
```

Comment:
Should we hard-block here (e.g. make a chan and block on it) rather than retry ?